### PR TITLE
Devops merge branch

### PR DESCRIPTION
**Pull request by matdenkas**
Helpful commands:
python3 tools/init_db.py # inits db
python3 src/semaphoreRunner.py -d test_dspec.json -p 202309051100 # runs the demo test

### CODE REVIEW

**Comment by matdenkas** [comment]
https://github.com/orgs/conrad-blucher-institute/projects/17/views/2?pane=issue&itemId=47898966

We are aware of the test failing in outputs here is its ticket

File: .vscode/launch.json
Comment (bestrada33) [inline .vscode/launch.json:1]: Should this be committed?

File: .vscode/tasks.json
Comment (bestrada33) [inline .vscode/tasks.json:1]: Should this be committed?

File: src/tests/test_SQLAlchemyORM.py [hunk 1]
```diff
@@ -108,48 +102,46 @@ def test_input_table_with_interval_6min():
...
     sqlorm.insert_input(series)
     result = sqlorm.select_input(seriesDescription, timeDescription)
 
     # Remove the invalid row to check that the db removed it as well
-    series.data.remove(Input('Apple', 'cherry', datetime(2001, 1, 1, 1, 0, 0, 0), datetime(2001, 1, 1, 8, 1, 0, 0), '12', '12'),)
+    series.data.remove(Input('1.5', 'meter', datetime(2001, 1, 1, 1, 0, 0, 0), datetime(2001, 1, 1, 8, 1, 0, 0), '12', '12'),)
     assert result.data == result.data
 
 
 def test_output_table():
```
Comment (bestrada33) [inline src/tests/test_SQLAlchemyORM.py:132]: This gives a bug right now. Just making note of it.

**Review by lovelysandlonelys — COMMENTED** [review]
everything looks/makes sense to me. However, I'm having an issue on my machine where I can't initialize the database inside the containers: 
image

**Comment by abeasley1722** [comment]
I'm having the same issue as Savannah

File: .dockerignore
Comment (abeasley1722) [inline .dockerignore:1]: LGTM. Just having the same issue as Savannah.

File: README.md [hunk 2]
```diff
@@ -27,6 +27,25 @@ pip install -r requirements.txt
...
+  - Python 3.10.5
+
+0. You will need to create a `.env` file in the root directory of the project. This file will contain the environment variables for the docker container. You can copy the contents of `env.dist` and replace the values with your own.
+1. Ensure that Docker Desktop is running on your machine
+2. Run `docker compose build` and `docker compose up` (run `docker compose up -d` to run in the background)
+3. SSH into semaphore container: `docker exec -it semaphore-core bash`
+4. Navigate and run command from inside the container as needed (e.g. `python3 tools/init_db.py` to init the DB)
+5. Database can be accessed using pgAdmin or VSCode Postgresql Explorer plugin
+ ![Register DB to pgAdmin](https://user-images.githubusercontent.com/7061990/268778360-2b92cdc0-19dd-48ae-853c-c52876f747d3.png)
+ ![Enter your database credentials to register to pgAdmin](https://user-images.githubusercontent.com/7061990/268778380-36b907c7-da08-4ba8-a232-0c7646dfbb82.png)
```
[the remark this answers is not in the document]
Comment (lovelysandlonelys) [reply README.md:44]: error above with password solved! there was an extra slash in the .dist file that has been removed

File: data/init_data/dataSource.csv [hunk 3]
```diff
@@ -0,0 +1,5 @@
+code,displayName,notes
+NOAATANDC,NOAA Tides and Currents,"NOAA Tides and Currents API"
+LIGHTHOUSE,Lighthouse,"The PD endpoint for Lighthouse"
+NDFD,NDFD,National Digital Forcast Database
```
Comment (abeasley1722) [inline data/init_data/dataSource.csv:4]: Should there be  " " here? And it's Forecast.

**Pull request merged**